jstests/ssl/x509/mkcert.py

```diff
@@ -479,31 +503,88 @@ def process_client_multivalue_rdn(cert):
     os.remove(rsa)
     os.remove(pem)
 
+def convert_ecdsa_key_to_pkcs8(ec_key_file, pkcs8_key_file):
+    """
+    Convert ECDSA key with explicit text header into a PEM-encoded PKCS#8 object
+    :param ec_key_file: name of file containing the PEM-encoded ECDSA key with explicit text header
+    :param pkcs8_key_file: name of file to contain PEM-encoded PKCS#8 object describing the ECDSA key
+    """
+    # wrap the PEM-encoded EC key with explicit text header into a PEM-encoded PKCS#8 object
+    pkcs8args = ['openssl', 'pkcs8', '-topk8', '-nocrypt', '-in', ec_key_file, '-out', pkcs8_key_file]
+    subprocess.check_call(pkcs8args)
+
+def process_ecdsa_cert(cert, pem, key, dest, filename, split_pem=True):
+    """Convert the ECDSA key and write the public/private key pair key into a .pem file, optionally writing the
+     public key to a .crt file and private key to a .key file"""
+
+    # copy the public key to a temp crt file
+    temp_cert_filename = tempfile.mkstemp()[1]
+    cert_filename = make_filename({'name': f"{filename}.crt"})
+    open(temp_cert_filename, 'wt').write(open(pem, 'rt').read())
+
+    # convert and create the temp key file
+    temp_key_filename = tempfile.mkstemp()[1]
+    convert_ecdsa_key_to_pkcs8(ec_key_file=key, pkcs8_key_file=temp_key_filename)
+
+    # combine public and private key into a .pem file with no comments
+    open(dest, 'wt').write(open(pem, 'rt').read() + open(temp_key_filename, 'rt').read())
+
+    if split_pem: # copy the temp files into .crt + .key files
+        cert_filename = make_filename({'name': f"{filename}.crt"})
+        open(cert_filename, 'wt').write(open(temp_cert_filename, 'rt').read())
+        key_filename = make_filename({'name': f"{filename}.key"})
+        open(key_filename, 'wt').write(open(temp_key_filename, 'rt').read())
```

Comment:
This line does not close either of the two files that it opens. Consider using shutil.copyfile instead (https://docs.python.org/2/library/shutil.html#shutil.copyfile).